types/mp4frag/index.d.ts

```diff
@@ -0,0 +1,183 @@
+// Type definitions for mp4frag 0.6
+// Project: https://github.com/kevinGodell/mp4frag
+// Definitions by: Ken Sanders <https://github.com/kensand>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+/// <reference types="node" />
+
+import { Transform } from 'stream';
+import 'node/buffer';
+
+export = Mp4Frag;
+
+/**
+ * An SegmentObject.
+ *
+ * An object with values
+ * <code>
+ * {
+ *     segment: null,
+ *     sequence: -1,
+ *     duration: -1;
+ *     timestamp: -1,
+ *     keyframe: true
+ * }
+ * </code
+ * is returned in erroneous conditions (before a segment has been generated, etc. Refer to individual function docs.
+ */
+interface SegmentObject {
+    segment?: Buffer;
+    sequence: number;
+    duration: number;
+    timestamp: number;
+    keyframe: boolean;
+}
+
+interface Mp4FragOptions {
+    /**
+     * Base name of files in m3u8 playlist. Affects the generated m3u8 playlist by naming file fragments.
+     *
+     * Must be set to generate m3u8 playlist. e.g. 'front_door'
+     */
+    hlsPlaylistBase?: string;
+
+    /**
+     * Number of segments to use in m3u8 playlist. Must be an integer ranging from 2 to 20.
+     *
+     * Defaults to 4.
+     */
+    hlsPlaylistSize?: number;
+
+    /**
+     * Number of extra segments to keep in memory. Must be an integer ranging from 0 to 10.
+     *
+     * Defaults to 0.
+     */
+    hlsPlaylistExtra?: number;
+
+    /**
+     * Indicates that m3u8 playlist should be generated after [initialization]{@link Mp4Frag#initialization}
+     * is created and before media segments are created.
+     *
+     * Defaults to true.
+     */
+    hlsPlaylistInit?: boolean;
+
+    /**
+     * Number of segments to keep in memory. Has no effect if using options.hlsPlaylistBase.
+     * Must be an integer ranging from 2 to 30.
+     *
+     * Defaults to 2.
+     */
+    segmentCount?: number;
+}
+
+declare class Mp4Frag extends Transform {
+    /**
+     * Construct a new Mp4Frag object.
+     *
+     * @throws If options.hlsPlaylistBase contains characters other than letters(a-zA-Z) and underscores(_).
+     */
+    constructor(options: Mp4FragOptions);
+
+    /**
+     * Returns the audio codec information as a <b>string</b>.
+     * Returns <b>null</b> if requested before [initialized event]{@link Mp4Frag#event:initialized}.
+     */
+    get audioCodec(): string | undefined;
```

Comment:
the owner opted to use implicit `null`, I'd probably stick to `null` here and in other places
https://github.com/kevinGodell/mp4frag/blob/master/index.js#L84
https://github.com/kevinGodell/mp4frag/blob/master/index.js#L88

Reply:
I considered that, and opted for undefined based on [this entry in the Typescript coding guidelines](https://github.com/Microsoft/TypeScript/wiki/Coding-guidelines#null-and-undefined) - does DefinitelyTyped have superseding guidelines for that? To be clear, I'm fine making that change, but if there's other guidelines I'm not aware of I'd like to check everything else in this PR against it too.

Reply:
I believe this guide was for case when writing code for TS source code. Declaration files should reflect native code, `null` is a same primitive as `undefined`. The implementation owner picked 'null' as return value (type).